### PR TITLE
Adding generic array support

### PR DESCRIPTION
**Pull request by ansharlubis**
Allowed frontend generic array support. Does not work with CPython right now.

### CODE REVIEW

File: integration_tests/CMakeLists.txt [hunk 1]
```diff
@@ -213,6 +213,7 @@ RUN(NAME vec_01              LABELS cpython llvm)
 RUN(NAME test_str_comparison LABELS cpython llvm)
 
 RUN(NAME generics_01         LABELS cpython llvm)
+RUN(NAME generics_array_01   LABELS llvm)
```
Comment (certik) [inline integration_tests/CMakeLists.txt:216]: Add a CPython test too:
```suggestion
RUN(NAME generics_array_01   LABELS cpython llvm)
```

**Comment by certik** [comment]
Excellent, great idea to get this working.

File: integration_tests/CMakeLists.txt (hunk 1 above)
Comment (ansharlubis) [reply integration_tests/CMakeLists.txt:216]: The problem is that a TypeVar object, in this case T, is not subscriptable in CPython.

Comment (certik) [reply integration_tests/CMakeLists.txt:216]: I see, that is indeed a problem.

I think we should probably design our own "TypeVar" that can handle subscripting, we'll add it to `ltypes` and we have to make sure exactly the same syntax works both in LPython and CPython.

File: integration_tests/generics_array_01.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
+from ltypes import TypeVar
+
+T = TypeVar('T')
+
+def f(lst: T[:], i: T) -> T:
```
Comment (czgdp1807) [inline integration_tests/generics_array_01.py:5]: A more interesting case would be to support `def f(lst: T[...], i: T) -> T`. We can do it later but we should keep in mind to support it.

**Comment by certik** [comment]
Thanks! I squashed and merged. I think this is fine to get started. We have to rework the syntax to get this working with CPython, and so on.

File: integration_tests/generics_array_01.py (hunk 2 above)
Comment (ansharlubis) [reply integration_tests/generics_array_01.py:5]: I haven't tried that out with CPython, but I bet the same issue as I mentioned above has to be solved first as well.

**Comment by ansharlubis** [comment]
Got it! I also think that defining our own TypeVar would be a good idea.

**Pull request merged**